### ISSUE TITLE
Duplicate output file crash in JobAccountant

### ISSUE THREAD

**Issue by amaltaro** [dmwm/WMCore#9011]
Related (or the same issue as) to:
https://github.com/dmwm/WMCore/issues/7305
https://github.com/dmwm/WMCore/issues/8598

given that it happens very rarely, we should simply drop one of the duplicate files (and fail the job) and move on. 

Traceback is
```
2019-02-11 10:08:44,938:139743250568960:INFO:AccountantWorker:Job 6049389 , handle successful job
2019-02-11 10:08:44,956:139743250568960:ERROR:AccountantWorker:Error while adding files to WMBS!
(IntegrityError) (1062, "Duplicate entry '/store/unmerged/RunIIFall17wmLHEGS/TT_AK8HT650_TuneCP5Up_PSweigh' for key 'lfn'") 'INSERT INTO wmbs_file_details (lfn, filesize, events,\n                                            first_event,
 merged)\n             VALUES (%s, %s, %s, %s,\n                     %s)' [('/store/unmerged/RunIIFall17wmLHEGS/TT_AK8HT650_TuneCP5Up_PSweights_13TeV-powheg-pythia8/LHE/93X_mc2017_realistic_v3-v1/80015/94ADC273-382D-E911-A9F0-6C3BE5B513E0.root', 63713622, 15539, 818734372L, 0), ('/store/unmerged/RunIIFall17wmLHEGS/TT_AK8HT650_TuneCP5Up_PSweights_13TeV-powheg-pythia8/GEN-SIM/93X_mc2017_realistic_v3-v1/80015/1E97C773-382D-E911-A9F0-6C3BE5B513E0.root', 45570792, 35, 818734372L, 0), ('/store/unmerged/logs/prod/2019/2/11/pdmvserv_task_TOP-RunIIFall17wmLHEGS-00279__v1_T_190121_150344_9094/TOP-RunIIFall17wmLHEGS-00279_0/0015/0/6bd58ebb-69ee-4678-ac0b-7d049a2bd8f1-689-0-logArchive.tar.gz', 0, 0, 818734372L, 0), ('/store/unmerged/RunIIFall17wmLHEGS/TT_AK8HT650_TuneCP5Up_PSweights_13TeV-powheg-pythia8/LHE/93X_mc2017_realistic_v3-v1/80015/367AB03C-362D-E911-A090-00259021A3D2.root', 98475277, 15539, 819060691L, 0), ('/store/unmerged/RunIIFall17wmLHEGS/TT_AK8HT650_TuneCP5Up_PSweights_13TeV-powheg-pythia8/GEN-SIM/93X_mc2017_realistic_v3-v1/80015/7A2BBB3C-362D-E911-A090-00259021A3D2.root', 55090223, 42, 819060691L, 0), ('/store/unmerged/logs/prod/2019/2/11/pdmvserv_task_TOP-RunIIFall17wmLHEGS-00279__v1_T_190121_150344_9094/TOP-RunIIFall17wmLHEGS-00279_0/0015/0/6bd58ebb-69ee-4678-ac0b-7d049a2bd8f1-710-0-logArchive.tar.gz', 0, 0, 819060691L, 0), ('/store/unmerged/RunIIFall17wmLHEGS/TT_AK8HT650_TuneCP5Up_PSweights_13TeV-powheg-pythia8/LHE/93X_mc2017_realistic_v3-v1/80015/56E2C43B-362D-E911-B2EC-00259021A306.root', 106709810, 15539, 819635634L, 0), ('/store/unmerged/RunIIFall17wmLHEGS/TT_AK8HT650_TuneCP5Up_PSweights_13TeV-powheg-pythia8/GEN-SIM/93X_mc2017_realistic_v3-v1/80015/CED2CF3B-362D-E911-B2EC-00259021A306.root', 35894168, 29, 819635634L, 0)  ... displaying 10 of 255 total bound parameter sets ...  ('/store/unmerged/RunIIFall17wmLHEGS/TT_AK8HT650_TuneCP5Up_PSweights_13TeV-powheg-pythia8/GEN-SIM/93X_mc2017_realistic_v3-v1/80016/969079DF-3F2D-E911-978C-0025904C540C.root', 43348647, 34, 1173971451L, 0), ('/store/unmerged/logs/prod/2019/2/10/pdmvserv_task_TOP-RunIIFall17wmLHEGS-00279__v1_T_190121_150344_9094/TOP-RunIIFall17wmLHEGS-00279_0/0016/0/d8a70541-d0ea-4bc3-8162-d7eee488f476-550-0-logArchive.tar.gz', 0, 0, 1173971451L, 0)]
2019-02-11 10:08:45,023:139743250568960:ERROR:BaseWorkerThread:Error in worker algorithm (1):
Backtrace:
  <WMComponent.JobAccountant.JobAccountantPoller.JobAccountantPoller object at 0x7f1882cf7c90> <@========== WMException Start ==========@>
Exception Class: AccountantWorkerException

Traceback: 
  File "/data/srv/wmagent/v1.1.18.patch2/sw/slc7_amd64_gcc630/cms/wmagent/1.1.18.patch2/lib/python2.7/site-packages/WMComponent/JobAccountant/AccountantWorker.py", line 831, in handleWMBSFiles
    transaction=self.existingTransaction())

  File "/data/srv/wmagent/v1.1.18.patch2/sw/slc7_amd64_gcc630/cms/wmagent/1.1.18.patch2/lib/python2.7/site-packages/WMCore/WMBS/MySQL/Files/Add.py", line 41, in execute
    conn=conn, transaction=transaction)

  File "/data/srv/wmagent/v1.1.18.patch2/sw/slc7_amd64_gcc630/cms/wmagent/1.1.18.patch2/lib/python2.7/site-packages/WMCore/Database/DBCore.py", line 160, in processData
    connection=connection, returnCursor=returnCursor))

  File "/data/srv/wmagent/v1.1.18.patch2/sw/slc7_amd64_gcc630/cms/wmagent/1.1.18.patch2/lib/python2.7/site-packages/WMCore/Database/MySQLCore.py", line 140, in executemanybinds
    returnCursor)

  File "/data/srv/wmagent/v1.1.18.patch2/sw/slc7_amd64_gcc630/cms/wmagent/1.1.18.patch2/lib/python2.7/site-packages/WMCore/Database/DBCore.py", line 113, in executemanybinds
    result = connection.execute(s, b)

  File "/data/srv/wmagent/v1.1.18.patch2/sw/slc7_amd64_gcc630/external/py2-sqlalchemy/0.9.6-comp/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 719, in execute
    return self._execute_text(object, multiparams, params)

  File "/data/srv/wmagent/v1.1.18.patch2/sw/slc7_amd64_gcc630/external/py2-sqlalchemy/0.9.6-comp/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 868, in _execute_text
    statement, parameters

  File "/data/srv/wmagent/v1.1.18.patch2/sw/slc7_amd64_gcc630/external/py2-sqlalchemy/0.9.6-comp/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 954, in _execute_context
    context)

  File "/data/srv/wmagent/v1.1.18.patch2/sw/slc7_amd64_gcc630/external/py2-sqlalchemy/0.9.6-comp/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1116, in _handle_dbapi_exception
    exc_info

  File "/data/srv/wmagent/v1.1.18.patch2/sw/slc7_amd64_gcc630/external/py2-sqlalchemy/0.9.6-comp/lib/python2.7/site-packages/sqlalchemy/util/compat.py", line 189, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb)

  File "/data/srv/wmagent/v1.1.18.patch2/sw/slc7_amd64_gcc630/external/py2-sqlalchemy/0.9.6-comp/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 924, in _execute_context
    context)

  File "/data/srv/wmagent/v1.1.18.patch2/sw/slc7_amd64_gcc630/external/py2-sqlalchemy/0.9.6-comp/lib/python2.7/site-packages/sqlalchemy/connectors/mysqldb.py", line 67, in do_executemany
    rowcount = cursor.executemany(statement, parameters)

  File "/data/srv/wmagent/v1.1.18.patch2/sw/slc7_amd64_gcc630/external/py2-mysqldb/1.2.4b4-comp/lib/python2.7/site-packages/MySQLdb/cursors.py", line 253, in executemany
    r = self._query('\n'.join([query[:p], ',\n'.join(q), query[e:]]))

  File "/data/srv/wmagent/v1.1.18.patch2/sw/slc7_amd64_gcc630/external/py2-mysqldb/1.2.4b4-comp/lib/python2.7/site-packages/MySQLdb/cursors.py", line 346, in _query
    rowcount = self._do_query(q)

  File "/data/srv/wmagent/v1.1.18.patch2/sw/slc7_amd64_gcc630/external/py2-mysqldb/1.2.4b4-comp/lib/python2.7/site-packages/MySQLdb/cursors.py", line 310, in _do_query
    db.query(q)
```

**Comment by vlimant** [comment]
could you maybe take care of the file/job removal so that the agent can go on

**Comment by amaltaro** [comment]
I have the feeling that this issue became more common in the production agents. Right now, even one of my testbed faced this issue (submit1). I'll try to get this sorted - even if not in an optimal way - by the coming week...

**Comment by amaltaro** [comment]
This has been happening quite often with the RelVal agent, where every week we have a burst of such uid clashes.

Given that RelVals are executed with TaskChain workflow types, this is a different than the one reported here: https://github.com/dmwm/WMCore/issues/9633 , even though the consequences are exactly the same.

We need to re-evaluate how the unique identifier is created and reduce the likelihood of having such clashes, even if that means a more complex UID algorithm (it all runs on the worker node anyways).

**Comment by khurtado** [comment]
Since CMSSW sets the GUIDs/uuids rather than us, this should be fixed with [#28551](https://github.com/cms-sw/cmssw/pull/28551) , [#37405](https://github.com/cms-sw/cmssw/pull/37405) and [#24246](https://github.com/cms-sw/cmssw/pull/24246) 

Closing this issue. Feel free to re-open if this incident is still observed.

**Comment by amaltaro** [comment]
Thanks Kenyi! And I confirm that it's been a few weeks that we don't hit such issue with the RelVal agent (which is usually testing the latest CMSSW releases).
